pages/signUp.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable react/no-unescaped-entities */
+
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+import { BaseButton } from "../components/shared/BaseButton";
+import { BaseInput } from "../components/shared/BaseInput";
+import { registerUser, NewUser } from "../api/user/register";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { useMutation } from "@tanstack/react-query";
+
+export interface IFormInput {
+    name: string,
+    email: string,
+    password: string,
+    repeatPassword: string,
+    userType: 'PATIENT' | 'DOCTOR'
+}
+
+export default function SignUpPage() {
+
+    const router = useRouter();
+    const { control, formState: { errors }, handleSubmit } = useForm<IFormInput>();
+
+    const [typeDoctor, setTypeDoctor] = useState(true)
+
+    const getRegisterData = (data: IFormInput) => {
+        const newUserData : NewUser = {
+            name: data.name,
+            email: data.email,
+            password: data.password,
+            user_type: typeDoctor ? 'DOCTOR' : 'PATIENT'
+        }
+        return newUserData;
+    }
+
+    const { mutate } = useMutation({
+        mutationFn: registerUser,
+        onSuccess: () => {
+            router.push('/login')
+        },
+      })
+
+    const onSubmit: SubmitHandler<IFormInput> = data => {
+        mutate(getRegisterData(data))
+    };
+
+    return (
+      <div className="w-full h-screen flex flex-col items-center justify-center">
+        <div className="w-[664px]">
+            <div className="w-full py-3 mb-10 flex flex-col items-center justify-center">
+                <h1 className="text-2xl font-semibold">Welcome to the doctor's app</h1>
+                <span className="mt-3 text-sm">Sign up to access unique features</span>
+            </div>
+            <form onSubmit={handleSubmit(onSubmit)}>
+                <div className="w-full flex flex-row items-center justify-between">
+                    <Controller
+                        control={control}
+                        rules={{ 
+                            required: {value: true, message: 'Name is required'} }}
+                        render={({ field: { onChange, value } }) => (
+                            <BaseInput label="Name" name='name' value={value} onChange={onChange} error={errors.name} errorMessage={errors.name?.message} />
+                        )}
+                        name="name"
+                    />
+                    <Controller
+                        control={control}
+                        rules={{ 
+                            required: {value: true, message: 'Email is required'}, 
+                            pattern: {
+                                value:
+                                /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```

Comment:
oh god, please either use a validator library or put this regex in a variable so we don't have to look at it here :D